### PR TITLE
WH: Add tag to metric rows_synced for all tables

### PR DESCRIPTION
**Pull request by psrikanth88**
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: warehouse/stats.go [hunk 1]
```diff
@@ -110,13 +111,20 @@ func (job *UploadJobT) generateUploadAbortedMetrics() {
...
+		if skipMetricTagForEachEventTable {
+			// club all event table metric tags under one tag to avoid too many tags
+			tableName = "others"
+		}
 	} else {
-		job.counterStat(`event_delivery`, tag{name: "tableName", value: strings.ToLower(tableName)}).Count(int(numEvents))
+		// record total events synced (ignoring additional row synced to event table for eg.track call)
+		// also ignore rows synced to rudder_discards as they are not full events but only discarded columns
+		if tableName != "rudder_discards" {
+			job.counterStat(`event_delivery`, tag{name: "tableName", value: strings.ToLower(tableName)}).Count(int(numEvents))
```
Comment (sivashanmukh) [inline warehouse/stats.go:126]: There is also rows_synced right after this, measuring the same number right? Why do we need both event_delivery and rows_synced?

Comment (psrikanth88) [reply warehouse/stats.go:126]: @sivashanmukh The rows_synced is to count total rows added to all tables in warehouse. `event_delivery` is to only count total events synced as one event can produce two rows etc.

Comment (SumanthPuram) [reply warehouse/stats.go:126]: We are using `event_delivery` for global metrics to track total delivered events across all destinations.

**Review by SumanthPuram — APPROVED** [review]

**Review by sivashanmukh — APPROVED** [review]
LGTM

**Pull request merged**